src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -374,6 +480,29 @@ public FilterAndTransform(string filterAndPayloadSpec, int startIdx, int endIdx,
                     }
                 }
 
+                Action<string, string, IEnumerable<KeyValuePair<string, string>>> writeEvent = null;
+                if (activityName != null && activityName.Contains("Activity"))
+                {
+                    MethodInfo writeEventMethodInfo = typeof(DiagnosticSourceEventSource).GetTypeInfo().GetDeclaredMethod(activityName);
```

Comment:
Should this be cached and reused for future accesses?  Or will this only ever really be used a few times at most?